File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        
        curSum = 0
        minLength = inf
        curLength = 0
        start = 0
        
        for i in range(len(nums)):
            curSum += nums[i]
            curLength += 1

            while curSum >= target:
                minLength = min(curLength, minLength)     
                curSum = curSum - nums[start]          
                curLength -= 1
                start += 1
            
        return 0 if minLength == inf else minLength

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {

    public int minSubArrayLen(int target, int[] nums) {
        
        int sum = 0;
        int start = 0;
        int result = Integer.MAX_VALUE;
        int count=0;

        for(int i=0; i<nums.length; i++){
            sum += nums[i];
            count++;    
            while(sum >= target){
                result = Math.min(result, count);                
                count--;                                
                sum = sum - nums[start]; 
                start++;
            }                        
        }                
        
        return result == Integer.MAX_VALUE ? 0 : result;
    }
}